### PR TITLE
Revert "Revert "Assignment Updates: Assign unit from course overview""

### PR DESCRIPTION
**Pull request by Erin007**
Reverts code-dot-org/code-dot-org#31578

### CODE REVIEW

**Comment by Erin007** [comment]
I reverted because of a failure in teacherSectionsReduxTest on staging. But I'm still fuzzy on how it passed locally: 
Screen Shot 2019-10-29 at 7 29 21 PM
and on Drone: 
Screen Shot 2019-10-29 at 9 04 05 PM

Maybe something about the order the tests run in?

File: apps/src/templates/teacherDashboard/teacherSectionsRedux.js [hunk 1]
```diff
@@ -972,7 +978,9 @@ export const sectionFromServerSection = serverSection => ({
   studentCount: serverSection.studentCount,
   code: serverSection.code,
   courseId: serverSection.course_id,
-  scriptId: serverSection.script ? serverSection.script.id : null,
+  scriptId: serverSection.script
+    ? serverSection.script.id
+    : serverSection.script_id || null,
```
Comment (Erin007) [inline apps/src/templates/teacherDashboard/teacherSectionsRedux.js:983]: It's tricky to tell from this Revert Revert, but the problem with the tests  was that null does not equal undefined. Making the  change above on line 983 and reversing below from #31578 to fix.
Screen Shot 2019-10-29 at 9 35 16 PM

**Comment by Erin007** [comment]
cc: @ajpal (yesterday's DOTD who caught the failure on staging) to close the loop.

**Review by dmcavoy — APPROVED** [review]
Your fix looks good! Thanks for tracking that down. 

Weird that it passed locally and in drone. Is there follow up we should do beyond updating this PR?

**Comment by Erin007** [comment]
> Your fix looks good! Thanks for tracking that down.
> 
> Weird that it passed locally and in drone. Is there follow up we should do beyond updating this PR?

Do you have something in mind? I'm not sure what that follow up would be.

**Pull request merged**